roles/ceph-monitor/tasks/main.yml

```diff
@@ -0,0 +1,102 @@
+---
+# we're storing the monitor_secret in a file on the first monitor node
+- name: check if mon_secrets file exists
+  stat: path=/var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+  run_once: true
+  delegate_to: "{{ groups['mons'][0] }}"
+  failed_when: false
+  register: mon_file_check
+
+- name: create monitor initial keyring
+  command: ceph-authtool /var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }} --create-keyring --name=mon. --gen-key --cap mon 'allow *'
+           creates=/var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+  run_once: true
+  delegate_to:  "{{ groups['mons'][0] }}"
+  when: mon_file_check.stat.exists == false
+
+- name: fetch contents of mon_secret file
+  slurp: path=/var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+  run_once: true
+  delegate_to: "{{ groups['mons'][0] }}"
+  register: mon_secret_file
+
+- name: copy mon_secret to all hosts
+  copy:
+    dest: /var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+    content: "{{ hostvars[groups['mons'][0]]['mon_secret_file']['content'] | b64decode }}"
+
+- name: set initial monitor key permissions
+  file: path=/var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+        mode=0600
+        owner=root
+        group=root
+
+- name: create monitor directory
+  file: path=/var/lib/ceph/mon/ceph-{{ ansible_hostname }}
+        state=directory
+        owner=root
+        group=root
+        mode=0755
+
+- name: ceph monitor mkfs
+  command: ceph-mon --mkfs -i {{ ansible_hostname }} --fsid {{ fsid_file.content | b64decode }} --keyring /var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
+           creates=/var/lib/ceph/mon/ceph-{{ ansible_hostname }}/keyring
+
+- name: activate monitor with upstart
+  file: path=/var/lib/ceph/mon/ceph-{{ ansible_hostname }}/{{ item }}
+        state=touch
+        owner=root
+        group=root
+        mode=0600
+  with_items:
+    - done
+    - upstart
+  changed_when: false
+
+- name: ensure ceph-mon service enabled and started
+  service: name=ceph-mon
+           state=started
+           enabled=yes
+           args=id={{ ansible_hostname }}
+
+# NOTE (leseb): wait for mon discovery and quorum resolution
+# the admin key is not instantanely created so we have to wait a bit
+- name: wait for client.admin key exists
+  wait_for: path=/etc/ceph/ceph.client.admin.keyring
+
+- name: create openstack pool
+  command: rados mkpool {{ item }}
+  with_items:
+    - "{{ ceph_monitor.openstack_block_storage_pool }}"
+  failed_when: false
```

Comment:
So when  you get different output for when the pool exists, what you can do is register: the task, and use the register variable in a changed_when statement.

```
register: poolout
changed_when: poolout.stdout | search('exists')
```

As for the rados one, may have to get more creative with the failed_when to search the output as well. Do these actions not cause anything to show up on the filesystem somewhere?